getcontext/tracing/trace.py

```diff
@@ -93,9 +98,59 @@ def evaluate(self):
             enforce_https=enforce_https(),
         )
 
-        return self.context_client.evaluations.result(
-            id=run_details.data.run_id, enforce_https=enforce_https()
+        results = self._poll_for_result(run_details.data.run_id)
+
+        if results.status == "failed":
+            raise EvaluationError(f"Evaluation failed:\n {results.as_dict()}")
+
+        failed_evaluation_msgs = self._parse_evaluation_results(results)
+        failed_msg = self._create_evaluation_fail_msg(failed_evaluation_msgs)
+
+        if failed_msg:
+            raise EvaluationException(failed_msg)
+
+        return results
+
+    def _parse_evaluation_results(self, results: EvaluationsRunResponse):
+        outcome_map = {
+            "positive": "passed",
+            "negative": "failed",
+            "partially_passed": "partially passed",
+            "inconclusive": "inconclusive",
```

Comment:
Would be nice if these magic strings were enums defined in our SDK! Perhaps a TODO?

Reply:
Yea could be good, at the same time this is more the sdk interpreting what negative and positive outcome mean with respect to forming an message for the user.